htlcswitch/link.go

```diff
@@ -256,9 +256,14 @@ type ChannelLinkConfig struct {
 
 	// TowerClient is an optional engine that manages the signing,
 	// encrypting, and uploading of justice transactions to the daemon's
-	// configured set of watchtowers.
+	// configured set of watchtowers for legacy channels.
 	TowerClient TowerClient
 
+	// AnchorTowerClient is an optional engine that manages the signing,
```

Comment:
Alternatively, this could be abstracted to use something similar to the session filter higher level function: the link calls something like `TowerClient(chanType)` and has that return the proper tower client. Once we also add more commitment types, this'll allow the link to be less cluttered by this selection logic.

Reply:
i was also thinking of another approach, where the peer selects the appropriate tower client to give to the link. since the two clients share the same interface, the link can be completely unaware as to which is the underlying. main benefit would be less code changes in the link and no function invocation overhead